### PR TITLE
Enable strict typing check in tests/ci

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable mypy checks in tests/ci directory

### CODE REVIEW

File: docker/test/style/process_style_check_result.py [hunk 1]
```diff
@@ -11,17 +11,19 @@ def process_result(result_folder):
...
     checks = (
-        ("header duplicates", "duplicate_output.txt"),
-        ("shellcheck", "shellcheck_output.txt"),
-        ("style", "style_output.txt"),
-        ("black", "black_output.txt"),
-        ("typos", "typos_output.txt"),
-        ("whitespaces", "whitespaces_output.txt"),
-        ("workflows", "workflows_output.txt"),
-        ("doc typos", "doc_spell_output.txt"),
+        "header duplicates",
```
Comment (rschu1ze) [inline docker/test/style/process_style_check_result.py:14]: Grammar: "duplicate includes" would be more correct.

(same in "docker/test/style/run.sh")

File: docker/test/style/process_style_check_result.py [hunk 2]
```diff
@@ -11,17 +11,19 @@ def process_result(result_folder):
...
+        "shellcheck",
+        "style",
+        "black",
+        "mypy",
+        "typos",
+        "whitespaces",
+        "workflows",
+        "doc typos",
```
Comment (rschu1ze) [inline docker/test/style/process_style_check_result.py:22]: "doc typos" --> "spelling"?

File: tests/ci/.mypy.ini [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[mypy]
+warn_no_return = False
+warn_unused_configs = True
+disallow_subclassing_any = True
+disallow_untyped_calls = False
```
Comment (rschu1ze) [inline tests/ci/.mypy.ini:5]: I did not check all options in this file, but at least `disallow_untyped_calls` is `False` by default already, i.e. can be removed (--> https://mypy.readthedocs.io/en/stable/config_file.html). Maybe take a look if there are more redundant options?

File: utils/check-style/check-mypy [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# The mypy supports pyproject.toml, but unfortunately it doesn't support it recursively
```
Comment (rschu1ze) [inline utils/check-style/check-mypy:3]: Just curious: I couldn't find any pyproject.toml files in our repository. Is "pyproj.toml" support even relevant to mypy? Such files don't contain Python code, right?

**Review by rschu1ze — APPROVED** [review]

File: docker/test/style/process_style_check_result.py (hunk 2 above)
Comment (Felixoid) [reply docker/test/style/process_style_check_result.py:22]: we already have typos... well, `docs spelling` looks good to me

File: tests/ci/.mypy.ini (hunk 3 above)
Comment (Felixoid) [reply tests/ci/.mypy.ini:5]: Here's everything included in `strict` mode https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict, but with explicit values. Explicit better than implicit, so I wouldn't change it.

Comment (Felixoid) [reply tests/ci/.mypy.ini:5]: Besides that, defaults can change from version to version

File: utils/check-style/check-mypy (hunk 4 above)
Comment (rschu1ze) [reply utils/check-style/check-mypy:3]: Uh.

**Comment by tavplubix** [comment]
Okay, merging with master helped, I suppose it was some kind of race condition in CI infrastructure

**Comment by Felixoid** [comment]
Indeed, the newest `clickhouse/style-check` requires `utils/check-style/check-mypy.sh` to be in the repo

**Comment by tavplubix** [comment]
@Felixoid, it breaks links in integration tests reports:
https://s3.amazonaws.com/clickhouse-test-reports/43132/fe8107e95f68f2571264e29eb755a2f979f1fa61/integration_tests__release__[3/4].html

**Comment by tavplubix** [comment]
Unfortunately, it cannot be easily reverted due to conflicts

**Comment by Felixoid** [comment]
Such a mess.. let me see what's changed and how to fix it.

**Comment by Felixoid** [comment]
Let's see if #43904 would fix it. If so, let's merge it, and then I'll take a closer look to the mess inside the types

**Pull request merged**